libs/data-mapper/src/lib/components/tree/SchemaTreeSearchbar.tsx

```diff
@@ -0,0 +1,182 @@
+  setFilteredDataTypes: (newDict: FilteredDataTypesDict) => void;
+}
+
+export const SchemaTreeSearchbar = ({ onSearch, onClear, filteredDataTypes, setFilteredDataTypes }: SchemaTreeSearchbarProps) => {
+  const intl = useIntl();
+  const styles = useStyles();
+
+  const [searchTerm, setSearchTerm] = useState<string>('');
+
+  const searchLoc = intl.formatMessage({
+    defaultMessage: 'Search',
+    description: 'Search',
+  });
+
+  const clearLoc = intl.formatMessage({
+    defaultMessage: 'Clear',
+    description: 'Clear',
+  });
+
+  const filterLoc = intl.formatMessage({
+    defaultMessage: 'Filter by data type',
+    description: 'Filter by data type',
+  });
+
+  const setSearchValue = (newValue: string) => {
+    setSearchTerm(newValue);
+    onChangeSearchValueDebounced(newValue);
+  };
+
+  const clearSearch = () => {
+    setSearchTerm('');
+    onClear();
+  };
+
+  const onChangeSearchValueDebounced = useDebouncedCallback(onSearch, [], searchDebounceDelay);
+
+  const onClickDataType = (dataType: NormalizedDataType | typeof arrayType | typeof selectAll, toggledOn: boolean) => {
+    const newFilteredDataTypes: FilteredDataTypesDict = { ...filteredDataTypes };
+
+    if (dataType === selectAll) {
+      Object.keys(filteredDataTypes).forEach((dataTypeKey) => {
+        newFilteredDataTypes[dataTypeKey as DataTypeFilterKeys] = toggledOn;
+      });
+    } else {
+      newFilteredDataTypes[dataType] = toggledOn;
+    }
+
+    setFilteredDataTypes({ ...newFilteredDataTypes });
+  };
+
+  const searchbarInteractions: JSX.Element = (
+    <>
+      {searchTerm && (
+        <Tooltip content={clearLoc} relationship="description">
+          <Button icon={<Dismiss20Regular />} appearance="subtle" onClick={clearSearch} aria-label={clearLoc} size="small" />
+        </Tooltip>
+      )}
+
+      <Menu persistOnItemClick>
+        <MenuTrigger>
+          <Tooltip content={filterLoc} relationship="description">
+            <Button icon={<Filter20Regular />} appearance="subtle" aria-label={filterLoc} size="small" />
+          </Tooltip>
+        </MenuTrigger>
+
+        <MenuPopover>
+          <MenuList>
+            <MenuItem onClick={() => onClickDataType(selectAll, !Object.values(filteredDataTypes).every((isFiltered) => !!isFiltered))}>
+              <Stack horizontal verticalAlign="center">
+                <Checkmark20Filled
+                  style={{
+                    visibility: Object.values(filteredDataTypes).some((isFiltered) => !isFiltered) ? 'hidden' : 'visible',
+                    marginRight: 2,
+                  }}
+                />
+                <Text style={{ color: tokens.colorBrandForegroundLink }}>Select all</Text>
+              </Stack>
+            </MenuItem>
+
+            {Object.entries(NormalizedDataType).flatMap(([dataTypeKey, dataTypeValue]) => {
+              const getMenuItemForDataType = (dataTypeKey: string, dataTypeValue: DataTypeFilterKeys) => {
```

Comment:
I think this is the spot where the icon is. Should we be using the colored icons or not, depending on whether they are selected? It feels weird that they are changing depending on selected

Reply:
Per the spec, yes